### DISCUSSION TITLE
Autofocus the search box on the home page on desktop.

### DISCUSSION THREAD

**Discussion by brentjanderson** [npm/feedback#339] [Website] [answered]
When I go to NPM's website the first thing I want to do is search for packages. I suspect that's the case for most developers. Wouldn't it be great if, when I go to https://npmjs.org/, it's ready for me to search? This would have the following benefits:

1. Those searching for packages would be used to going to NPMJS.org for that purpose.
2. If you are there to learn about NPM, the autofocus feature wouldn't detract from visiting the rest of the site.
3. This autofocus doesn't have to happen on any other page, so it wouldn't detract from reading docs etc.
4. This feature should be disabled if on a mobile device to prevent the keyboard from occluding the page.

Maybe this has been explored before, but I couldn't find existing discussions on the topic and it's something that I think of _every time_ I visit https://npmjs.org/. Thanks!

**Comment by ArthurKa** [comment, marked as answer]
After visiting https://npmjs.org just hit `Tab, Enter, Tab` and voila — you are in the serach box.

**Reply by MylesBorins** [reply]
I'll also add that the bar that pops down when you hit `tab` the first time offers a couple different option for quickly navigating the page.

**Comment by brentjanderson** [comment]
This is true, and it’s a great feature. Autofocus would be the last inch of optimization. Without looking at analytics for the home page, I suspect that most users hit the home page so that they can search for packages. Why not optimize for this use case unless it harms other users (e.g. screen readers, those looking to learn more about NPM as a whole).

With millions of developers relying on that search capability, that extra inch of optimization would be an improvement akin to shaving a moment off of a computer’s startup time. It may not matter in the short-run, but in aggregate it saves everyone a lot of time.
